holidays/countries/trinidad_and_tobago.py

```diff
@@ -0,0 +1,163 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomHinduHolidays, _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, OCT, NOV
+from holidays.constants import OPTIONAL, PUBLIC
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class TrinidadAndTobago(
+    HolidayBase, ChristianHolidays, HinduCalendarHolidays, InternationalHolidays, IslamicHolidays
+):
+    """Trinidad and Tobago holidays.
+
+    References:
+        * <https://otp.tt/trinidad-and-tobago/national-holidays-and-awards/>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Trinidad_and_Tobago>
+        * <https://en.wikipedia.org/wiki/Trinidad_and_Tobago_Carnival>
+        * <https://en.wikipedia.org/wiki/Indian_Arrival_Day#Trinidad_and_Tobago>
+        * <https://www.timeanddate.com/holidays/trinidad/eid-al-fitr>
+        * <https://www.timeanddate.com/holidays/trinidad/diwali>
+    """
+
+    country = "TT"
+    default_language = "en_TT"
+    supported_categories = (OPTIONAL, PUBLIC)
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    supported_languages = ("en_TT", "en_US")
+    # Trinidad and Tobago gained independence on August 31, 1962.
+    start_year = 1963
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=TrinidadAndTobagoHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=TrinidadAndTobagoIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("New Year's Day"))
```

Comment:
`dts_observed.add( ... )` should be add for any entry that can falls on Sunday
```suggestion
        dts_observed = set()

        # New Year's Day.
        dts_observed.add(self._add_new_years_day(tr("New Year's Day")))
```